queenbee/job/job.py

```diff
@@ -44,6 +44,23 @@ class Job(BaseModel):
     )
 
 
+class JobStatusEnum(str, Enum):
+    """Enumaration of allowable status strings"""
+
+    # The job has been created
+    created = 'Created'
+    # The job folder is being prepared for execution and runs are being scheduled
+    pre_processing = 'Pre-Processing'
```

Comment:
What do you think about keeping them all lowercase and punctuation free? It seems easy to forget what the convention is in the case of doing a direct string comparison. (E.g is it `preprocessing` or `Pre-Processing`?) But I can be convinced otherwise!

Reply:
Am I right in thinking you perceive this as an issue for developers using the Python Queenbee library? 

My thinking is that once you start using `Enum` you should be using the Enum class for string comparison instead of direct string comparison. This means that the enum "raw" value is different from the "conceptual" value it represents in your code. This should in theory make things more robust.

```python
status_string = 'Pre-Processing'

# Brittle
# Direct string comparison
if status_string == 'Pre-Processing':
  do_stuff()

# More robust
# Enum value based comparison
if status_string == JobStatusEnum.pre_processing:
  do_stuff()
```

Reply:
I agree with the comparison but there is another reason that I think removing the `-` is a good idea. When enums are translated to C# through (Pydantic -> OpenAPI -> Code generator -> Queenbee .NET) it only takes the value - not the name during the translation. Foe the value to be value we should remove the `-` so either postprocessing or PostProcessing are good options. @MingboPeng, let me know if this has changed from the last time that we talked about.

Also what happened to `canceled` status?

Reply:
Yep, I was thinking of the python library only. And I agree that one should be using the more robust enum value comparison. My point was that, in either case, I (a hypothetical consumer) need to remember what the convention is for the string literal that is put into `status_string`.

@mostaphaRoudsari brings up a good point about the code generation.

Reply:
I am going to wait for @MingboPeng to comment on this before merging but I just ran the OpenAPI generator for the Queenbee C# SDK repo and `Enums` clean up the string for the value name and render like this:

```c#
/* 
 * Queenbee Schema with Inheritance
 *
 * Documentation for Queenbee schema.
 *
 * Contact: info@ladybug.tools
 * Generated by: https://github.com/openapitools/openapi-generator.git
 */

using System;
using System.Linq;
using System.IO;
using System.Text;
using System.Text.RegularExpressions;
using System.Collections;
using System.Collections.Generic;
using System.Collections.ObjectModel;
using System.Runtime.Serialization;
using Newtonsoft.Json;
using Newtonsoft.Json.Converters;
using System.ComponentModel.DataAnnotations;


namespace QueenbeeSDK
{
    /// <summary>
    /// Enumaration of allowable status strings
    /// </summary>
    /// <value>Enumaration of allowable status strings</value>
    
    [JsonConverter(typeof(StringEnumConverter))]
    
    public enum RunStatusEnum
    {
        /// <summary>
        /// Enum Created for value: Created
        /// </summary>
        [EnumMember(Value = "Created")]
        Created = 1,

        /// <summary>
        /// Enum Scheduled for value: Scheduled
        /// </summary>
        [EnumMember(Value = "Scheduled")]
        Scheduled = 2,

        /// <summary>
        /// Enum Running for value: Running
        /// </summary>
        [EnumMember(Value = "Running")]
        Running = 3,

        /// <summary>
        /// Enum PostProcessing for value: Post-Processing
        /// </summary>
        [EnumMember(Value = "Post-Processing")]
        PostProcessing = 4,

        /// <summary>
        /// Enum Failed for value: Failed
        /// </summary>
        [EnumMember(Value = "Failed")]
        Failed = 5,

        /// <summary>
        /// Enum Succeeded for value: Succeeded
        /// </summary>
        [EnumMember(Value = "Succeeded")]
        Succeeded = 6,

        /// <summary>
        /// Enum Unknown for value: Unknown
        /// </summary>
        [EnumMember(Value = "Unknown")]
        Unknown = 7

    }

}
```

So it still seems ok for us to use `Post-Processing` as a value. It's worth noting I have an interest in keeping it that way for 2 reasons:
- An enum value is a value that shouldn't be concerned with how code handles it (ie: we should be allowed to have `illegal` string types in an enum)
- The database is currently filled with strings in the format above and I'd rather not run another migration 🙃

Reply:
Yes, OpenAPI generator will remove all special characters during the translation. But I would vote for the "PostProcessing" so that it keeps consistent across different languages.
CSharp:
```csharp
JobStatusEnum.PostProcessing 
```

Python:
```python
JobStatusEnum.PostProcessing 
```